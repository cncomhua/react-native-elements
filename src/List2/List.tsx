export { default as Item } from './ListItem';
export { default as Content } from './ListContent';
export { default as Accordion } from './ListAccordion';
export { default as Title } from './ListTitle';
export { default as SubTitle } from './ListSubtitle';
export { default as Section } from './ListSection';
