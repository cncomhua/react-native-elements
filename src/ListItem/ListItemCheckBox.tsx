import React from 'react';
import { StyleSheet } from 'react-native';
import CheckBox, { CheckBoxProps } from '../Checkbox';
import { RneFunctionComponent } from '../helpers';

export const ListItemCheckBox: RneFunctionComponent<CheckBoxProps> = ({
  containerStyle,
  ...props
}) => {
  return (
    <CheckBox
      {...props}
      containerStyle={StyleSheet.flatten([styles.container, containerStyle])}
    />
  );
};

const styles = StyleSheet.create({
  container: {
    margin: 0,
    marginRight: 0,
    marginLeft: 0,
    padding: 0,
  },
});

ListItemCheckBox.displayName = 'ListItemCheckBox';
