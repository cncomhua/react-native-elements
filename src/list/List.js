import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';
import ViewPropTypes from '../config/ViewPropTypes';

class List extends React.PureComponent {
  render() {
    const { children, containerStyle, ...attributes } = this.props;
    const styles = this.context.theme.list;

    return (
      <View
        style={[styles.listContainer, containerStyle && containerStyle]}
        {...attributes}
      >
        {children}
      </View>
    );
  }
}

List.propTypes = {
  children: PropTypes.any,
  containerStyle: ViewPropTypes.style,
};

List.contextTypes = {
  theme: PropTypes.object.isRequired,
};

export default List;
