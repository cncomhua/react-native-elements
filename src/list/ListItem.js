import PropTypes from 'prop-types';
import React from 'react';
import {
  Platform,
  StyleSheet,
  Switch,
  TouchableOpacity,
  View,
} from 'react-native';
import Avatar from '../avatar/Avatar';
import Badge from '../badge/badge';
import CheckBox from '../checkbox/CheckBox';
import Icon from '../icons/Icon';
import Text from '../text/Text';
import TextInput from '../input/Input';
import ViewPropTypes from '../config/ViewPropTypes';

console.disableYellowBox = true;

const IOS_BLUE = '#007AFF';
const ANDROID_SECONDARY = 'rgba(0, 0, 0, 0.54)';

const ListItem = props => {
  const {
    title,
    titleProps,
    subtitle,
    subtitleProps,
    containerStyle,
    component,
    icon,
    avatar,
    leftElement,
    rightElement,
    rightTitle,
    rightTitleProps,
    textInputProps,
    switchProps,
    checkBoxProps,
    badgeProps,
    disclosure,
    centerContainerStyle,
    checkmark,
    ...attributes
  } = props;
  delete attributes.avatar;

  const { onPress, onLongPress } = props;
  let Component =
    component || (onPress || onLongPress ? TouchableOpacity : View);

  return (
    <Component {...attributes} onPress={onPress}>
      <PadView style={[styles.container, containerStyle]}>
        {renderNode(leftElement)}
        {renderIcon(icon)}
        {renderNode(
          icon,
          {
            color: Platform.OS === 'ios' ? null : ANDROID_SECONDARY,
            size: 24,
            ...icon,
            containerStyle: [styles.iconContainer, icon && icon.containerStyle],
          },
          null,
          Icon
        )}
        {renderNode(
          avatar,
          { width: 40, height: 40, rounded: true, ...avatar },
          null,
          Avatar
        )}
        <View
          style={[
            styles.centerContainer,
            textInputProps && { flex: 0 },
            centerContainerStyle,
          ]}
        >
          {renderNode(title, titleProps, styles.title)}
          {renderNode(subtitle, subtitleProps, styles.subtitle)}
        </View>
        {renderNode(rightTitle, rightTitleProps, styles.rightTitle)}
        {textInputProps && (
          <TextInput
            {...textInputProps}
            style={[styles.textInput, textInputProps && textInputProps.style]}
            containerStyle={[
              styles.textInputContainer,
              textInputProps && textInputProps.containerStyle,
            ]}
          />
        )}
        {switchProps && <Switch {...switchProps} />}
        {checkBoxProps && <CheckBox {...checkBoxProps} />}
        {badgeProps && <Badge {...badgeProps} />}
        {renderNode(rightElement)}
        {checkmark && Checkmark}
        {disclosure && Disclosure}
      </PadView>
    </Component>
  );
};

const styles = StyleSheet.create({
  container: {
    ...Platform.select({
      ios: {
        padding: 14,
      },
      android: {
        padding: 16,
      },
    }),
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: 'white',
  },
  iconContainer: {
    width: 40,
    height: 40,
    borderRadius: 20,
  },
  title: {
    ...Platform.select({
      ios: {
        fontSize: 17,
      },
      android: {
        fontSize: 16,
      },
    }),
  },
  subtitle: {
    ...Platform.select({
      ios: {
        fontSize: 15,
      },
      android: {
        color: ANDROID_SECONDARY,
        fontSize: 14,
      },
    }),
  },
  centerContainer: {
    flex: 1,
    justifyContent: 'center',
  },
  textInput: {
    flex: 1,
    textAlign: 'right',
  },
  textInputContainer: {
    borderBottomWidth: 0,
  },
  rightTitle: {
    ...Platform.select({
      ios: {
        fontSize: 17,
      },
      android: {
        fontSize: 16,
      },
    }),
    color: ANDROID_SECONDARY,
  },
});

const Disclosure = (
  <Icon
    type={Platform.OS === 'ios' ? 'ionicon' : 'material'}
    name={Platform.OS === 'ios' ? 'ios-arrow-forward' : 'keyboard-arrow-right'}
    size={16}
    color="#D1D1D6"
  />
);

const Checkmark = (
  <Icon
    type={Platform.OS === 'ios' ? 'ionicon' : 'material'}
    name={Platform.OS === 'ios' ? 'ios-checkmark' : 'check'}
    size={Platform.OS === 'ios' ? 34 : 20}
    color={Platform.OS === 'ios' ? IOS_BLUE : ANDROID_SECONDARY}
  />
);

ListItem.propTypes = {
  containerStyle: ViewPropTypes.style,
  centerContainerStyle: ViewPropTypes.style,
  component: PropTypes.element,
  onPress: PropTypes.func,
  onLongPress: PropTypes.func,
  title: PropTypes.node,
  subtitle: PropTypes.node,
  titleProps: PropTypes.object,
  subtitleProps: PropTypes.object,
  icon: PropTypes.node,
  avatar: PropTypes.node,
  leftElement: PropTypes.element,
  rightElement: PropTypes.element,
  rightTitle: PropTypes.node,
  rightTitleProps: PropTypes.object,
  textInputProps: PropTypes.object,
  switchProps: PropTypes.object,
  checkBoxProps: PropTypes.object,
  badgeProps: PropTypes.object,
  disclosure: PropTypes.bool,
  checkmark: PropTypes.bool,
};

const PadView = ({ children, pad = 16, ...props }) => {
  const childrens = React.Children.toArray(children);
  const length = childrens.length;
  return (
    <View {...props}>
      {React.Children.map(
        childrens,
        (child, index) =>
          child && [child, index !== length - 1 && <View width={pad} />]
      )}
    </View>
  );
};

const renderIcon = content =>
  content == null ? null : React.isValidElement(content) ? (
    content
  ) : (
    <Icon
      color={Platform.OS === 'ios' ? null : ANDROID_SECONDARY}
      size={24}
      {...content}
      containerStyle={[styles.iconContainer, content && content.containerStyle]}
    />
  );

const renderNode = (content, props, style, Component = Text) =>
  content == null ? null : React.isValidElement(content) ? (
    content
  ) : (
    <Component {...props} style={[style, props && props.style]}>
      {Component === Text && content}
    </Component>
  );

export default ListItem;
