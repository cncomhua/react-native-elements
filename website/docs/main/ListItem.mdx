---
id: listitem

title: ListItem

slug: /listitem
---

import Usage from './usage/ListItem/ListItem.mdx';

import Props from './props/ListItem.mdx';

import { ListItem } from 'react-native-elements';

import Tabs from '@theme/Tabs';

import TabItem from '@theme/TabItem';

ListItems are used to display rows of information, such as a contact list, playlist, or menu.

They are very customizable and can contain switches, avatars, badges, icons, and more.

## Components

- [ListItem.Accordion](#listitemaccordion)
  This allows making a accordion list which can show/hide content.
- [ListItem.ButtonGroup](#listitembuttongroup)
  This allows adding ButtonGroup to the ListItem.
  This, Receives all [ButtonGroup](buttongroup#props) props.
- [ListItem.CheckBox](#listitemcheckbox)
  This allows adding CheckBox to the ListItem.
  This, Receives all [CheckBox](checkbox#props) props.
- [ListItem.Chevron](#listitemchevron)
  This allows adding a Chevron Icon(arrow) to the ListItem.
  This, Receives all [Icon](icon#props) props.
- [ListItem.Content](#listitemcontent)
  This allows adding content to the ListItem.
  This, Receives all [View](https://reactnative.dev/docs/view#props) props.
- [ListItem.Input](#listiteminput)
  This allows adding an Text Input within the ListItem.
  This, Receives all [Input](Input.mdx#props) props.
- [ListItem.Subtitle](#listitemsubtitle)
  This allows adding SubTitle to the ListItem.
  This, Receives all [Text](text#props) props.
- [ListItem.Swipeable](#listitemswipeable)
  We offer a special kind of ListItem which is swipeable from both ends and allows users select an event.
- [ListItem.Title](#listitemtitle)
  This allows adding Title to the ListItem.
  This, Receives all [Text](text#props) props.

## Usage

<Usage />

## Props

<Props />
