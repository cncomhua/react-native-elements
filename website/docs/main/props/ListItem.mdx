### ListItem

- [Component](#component)
- [ViewComponent](#viewcomponent)
- [bottomDivider](#bottomdivider)
- [children](#children)
- [containerStyle](#containerstyle)
- [disabledStyle](#disabledstyle)
- [linearGradientProps](#lineargradientprops)
- [pad](#pad)
- [topDivider](#topdivider)

### ListItem.Accordion

- [animation](#animation)
- [content](#content)
- [expandIcon](#expandicon)
- [icon](#icon)
- [isExpanded](#isexpanded)
- [noIcon](#noicon)
- [noRotation](#norotation)

### ListItem.ButtonGroup

None

### ListItem.CheckBox

None

### ListItem.Chevron

None

### ListItem.Content

- [right](#right)

### ListItem.Input

None

### ListItem.Subtitle

- [right](#right)

### ListItem.Swipeable

- [leftContent](#leftcontent)
- [leftStyle](#leftstyle)
- [leftWidth](#leftwidth)
- [onLeftSwipe](#onleftswipe)
- [onRightSwipe](#onrightswipe)
- [rightContent](#rightcontent)
- [rightStyle](#rightstyle)
- [rightWidth](#rightwidth)

### ListItem.Title

- [right](#right)

## Reference

### ListItem

#### Component

Replace element with custom element.

| Type            | Default |
| --------------- | ------- |
| React Component | None    |

---

#### ViewComponent

Container for linear gradient.

| Type            | Default |
| --------------- | ------- |
| React Component | None    |

---

#### bottomDivider

Add divider at the bottom of the list item.

| Type    | Default |
| ------- | ------- |
| boolean | None    |

---

#### children

Either children or a render prop that receives a boolean reflecting whether

the component is currently pressed.

Add enclosed children.

| Type | Default |
| ---- | ------- |
| any  | None    |

---

#### containerStyle

Additional main container styling.

| Type               | Default |
| ------------------ | ------- |
| View style(Object) | None    |

---

#### disabledStyle

Specific styling to be used when list item is disabled.

| Type               | Default |
| ------------------ | ------- |
| View style(Object) | None    |

---

#### linearGradientProps

Props for linear gradient component.

| Type | Default |
| ---- | ------- |
| any  | None    |

---

#### pad

Adds spacing between the leftComponent, the title component & right component.

| Type   | Default |
| ------ | ------- |
| number | None    |

---

#### topDivider

Add divider at the top of the list item.

| Type    | Default |
| ------- | ------- |
| boolean | None    |

---

### ListItem.Accordion

#### animation

Decide whether to show animation.

| Type              | Default |
| ----------------- | ------- |
| Boolean or Object | {       |

      duration: 350,
      type: 'timing',
    } |

---

#### content

Similar to ListItem's child.

| Type            | Default |
| --------------- | ------- |
| React.ReactNode | None    |

---

#### expandIcon

Icon when Accordion is expanded, if not provided `icon` will be rotated 180deg.

| Type | Default |
| ---- | ------- |
| any  | None    |

---

#### icon

Icon for unexpanded Accordion.

| Type | Default |
| ---- | ------- |
| any  | None    |

---

#### isExpanded

Decide if Accordion is Expanded.

| Type    | Default |
| ------- | ------- |
| boolean | None    |

---

#### noIcon

Don't show accordion icon.

| Type    | Default |
| ------- | ------- |
| boolean | None    |

---

#### noRotation

Don't rotate when Accordion is expanded.

| Type    | Default |
| ------- | ------- |
| boolean | None    |

---

### ListItem.ButtonGroup

None

### ListItem.CheckBox

None

### ListItem.Chevron

None

### ListItem.Content

#### right

| Type    | Default |
| ------- | ------- |
| boolean | None    |

---

### ListItem.Input

None

### ListItem.Subtitle

#### right

| Type    | Default |
| ------- | ------- |
| boolean | None    |

---

### ListItem.Swipeable

#### leftContent

Left Content.

| Type            | Default |
| --------------- | ------- |
| React.ReactNode | None    |

---

#### leftStyle

Style of left container.

| Type               | Default |
| ------------------ | ------- |
| View style(Object) | None    |

---

#### leftWidth

Width to swipe left.

| Type   | Default         |
| ------ | --------------- |
| number | ScreenWidth / 3 |

---

#### onLeftSwipe

Function to call when user swipes left.

| Type     | Default |
| -------- | ------- |
| Function | None    |

---

#### onRightSwipe

Function to call when user swipes right.

| Type     | Default |
| -------- | ------- |
| Function | None    |

---

#### rightContent

Right Content.

| Type            | Default |
| --------------- | ------- |
| React.ReactNode | None    |

---

#### rightStyle

Style of right container.

| Type               | Default |
| ------------------ | ------- |
| View style(Object) | None    |

---

#### rightWidth

Width to swipe right.

| Type   | Default         |
| ------ | --------------- |
| number | ScreenWidth / 3 |

---

### ListItem.Title

#### right

Add right title.

| Type    | Default |
| ------- | ------- |
| boolean | None    |

---
