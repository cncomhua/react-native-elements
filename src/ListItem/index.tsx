import { withTheme } from '../config';
import { ListItemBaseProps, ListItemBase } from './ListItemBase';
import ListItemContent from './ListItemContent';
import ListItemChevron from './ListItemChevron';
import ListItemInput from './ListItemInput';
import ListItemCheckBox from './ListItemCheckBox';
import ListItemButtonGroup from './ListItemButtonGroup';
import ListItemTitle from './ListItemTitle';
import ListItemSubtitle from './ListItemSubtitle';
import ListItemSwipeable, { ListItemSwipeableProps } from './ListItemSwipeable';
import ListItemAccordion, { ListItemAccordionProps } from './ListItemAccordion';
import { RneFunctionComponent } from '../helpers';

export type ListItemProps = RneFunctionComponent<ListItemBaseProps> & {
  Accordion: typeof ListItemAccordion;
  Chevron: typeof ListItemChevron;
  Content: typeof ListItemContent;
  Input: typeof ListItemInput;
  Title: typeof ListItemTitle;
  Subtitle: typeof ListItemSubtitle;
  Swipeable: typeof ListItemSwipeable;
  CheckBox: typeof ListItemCheckBox;
  ButtonGroup: typeof ListItemButtonGroup;
};

export const ListItem: ListItemProps = Object.assign(ListItemBase);

export type { ListItemAccordionProps, ListItemSwipeableProps };

const ThemedListItem = Object.assign(withTheme(ListItem, 'ListItem'), {
  Accordion: ListItemAccordion,
  Chevron: ListItemChevron,
  Content: ListItemContent,
  Input: ListItemInput,
  Title: ListItemTitle,
  Subtitle: ListItemSubtitle,
  Swipeable: ListItemSwipeable,
  CheckBox: ListItemCheckBox,
  ButtonGroup: ListItemButtonGroup,
});

export default ThemedListItem;
