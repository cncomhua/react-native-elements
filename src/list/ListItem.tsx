import { withTheme } from '../config';
import { ListItemProps, ListItemBase } from './ListItemBase';
import ListItemContent from './ListItemContent';
import ListItemChevron from './ListItemChevron';
import ListItemInput from './ListItemInput';
import ListItemCheckBox from './ListItemCheckBox';
import ListItemButtonGroup from './ListItemButtonGroup';
import ListItemTitle from './ListItemTitle';
import ListItemSubtitle from './ListItemSubtitle';
import ListItemAccordion from './ListItemAccordion';
import { RneFunctionComponent } from '../helpers';

interface ListItem extends RneFunctionComponent<ListItemProps> {
  Accordion: typeof ListItemAccordion;
  Chevron: typeof ListItemChevron;
  Content: typeof ListItemContent;
  Input: typeof ListItemInput;
  Title: typeof ListItemTitle;
  Subtitle: typeof ListItemSubtitle;
  CheckBox: typeof ListItemCheckBox;
  ButtonGroup: typeof ListItemButtonGroup;
}

const ListItem: ListItem = Object.assign(ListItemBase);

export { ListItem };

const ThemedListItem = Object.assign(withTheme(ListItem, 'ListItem'), {
  Accordion: ListItemAccordion,
  Chevron: ListItemChevron,
  Content: ListItemContent,
  Input: ListItemInput,
  Title: ListItemTitle,
  Subtitle: ListItemSubtitle,
  CheckBox: ListItemCheckBox,
  ButtonGroup: ListItemButtonGroup,
});

export default ThemedListItem;
