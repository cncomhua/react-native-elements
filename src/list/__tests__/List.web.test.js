import React from 'react';
import { Text, TouchableHighlight, View } from 'react-native';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import parseSamples, {
  assignRef,
  genRefId,
} from 'enzyme-styleguidist-sample-parser';
import options, {
  snapShot,
  buildJsxForGuideMethod,
  ensureCalled,
  onlyEnsureCalled,
  onlySnapshots,
} from '../../../samples';
import List from '../List';
import ListItem from '../ListItem';

const children = [
  <ListItem key="v0" title="item 1" />,
  <ListItem key="v1" title="item 2" />,
  <ListItem key="v2" title="item 3" />,
];

const props = {
  'no props': {
    component: List,
    enzyme: {
      tests: onlySnapshots,
    },
    styleguidist: {},
  },
  containerStyle: {
    component: List,
    children,
    props: {
      containerStyle: { borderWidth: 5 },
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
};

const samples = {
  props,
};

parseSamples(
  {
    Lists: { sectionComponents: { List: { samples } } },
  },
  options
);

describe('List component', () => {
  it('should render without issues', () => {
    const component = shallow(<List />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render children', () => {
    const component = shallow(
      <List containerStyle={{ backgroundColor: 'peru' }}>
        <ListItem />
      </List>
    );

    expect(component.find('ListItem').length).toBe(1);
  });
});
