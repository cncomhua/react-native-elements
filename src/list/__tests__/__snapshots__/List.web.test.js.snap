// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component should render without issues 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderColor": "#bbb",
      "borderTopWidth": 1,
      "marginTop": 20,
    }
  }
/>
`;

exports[`Lists: List: Props Lists: List: [ prop: containerStyle ] enzyme-shallow: snapshot: Lists: List: [ prop: containerStyle ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderColor": "#bbb",
      "borderTopWidth": 1,
      "borderWidth": 5,
      "marginTop": 20,
    }
  }
>
  <ListItem
    chevronColor="#bdc6cf"
    disabled={false}
    hideChevron={false}
    key="v0"
    leftIconUnderlayColor="white"
    rightIcon={
      Object {
        "name": "chevron-right",
      }
    }
    rightTitleNumberOfLines={1}
    roundAvatar={false}
    subtitleNumberOfLines={1}
    switchButton={false}
    textInputEditable={true}
    title="item 1"
    titleNumberOfLines={1}
    underlayColor="white"
  />
  <ListItem
    chevronColor="#bdc6cf"
    disabled={false}
    hideChevron={false}
    key="v1"
    leftIconUnderlayColor="white"
    rightIcon={
      Object {
        "name": "chevron-right",
      }
    }
    rightTitleNumberOfLines={1}
    roundAvatar={false}
    subtitleNumberOfLines={1}
    switchButton={false}
    textInputEditable={true}
    title="item 2"
    titleNumberOfLines={1}
    underlayColor="white"
  />
  <ListItem
    chevronColor="#bdc6cf"
    disabled={false}
    hideChevron={false}
    key="v2"
    leftIconUnderlayColor="white"
    rightIcon={
      Object {
        "name": "chevron-right",
      }
    }
    rightTitleNumberOfLines={1}
    roundAvatar={false}
    subtitleNumberOfLines={1}
    switchButton={false}
    textInputEditable={true}
    title="item 3"
    titleNumberOfLines={1}
    underlayColor="white"
  />
</View>
`;

exports[`Lists: List: Props Lists: List: [ prop: no props ] enzyme-mount: snapshot: Lists: List: [ prop: no props ] enzyme-mount: snapshot 1`] = `
<List>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "#bbb",
        "borderTopWidth": 1,
        "marginTop": 20,
      }
    }
  >
    <div
      className="rn-alignItems-1oszu61 rn-backgroundColor-44z8sh rn-borderTopColor-gv7t1l rn-borderRightColor-1itbhdy rn-borderBottomColor-1u5ghgs rn-borderLeftColor-592z09 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-5kkj8d rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1pxmb3b rn-flexBasis-7vfszb rn-flexDirection-eqz5dr rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-marginTop-1x0uki6 rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
    />
  </View>
</List>
`;

exports[`Lists: List: Props Lists: List: [ prop: no props ] enzyme-render: snapshot: Lists: List: [ prop: no props ] enzyme-render: snapshot 1`] = `
<div
  class="rn-alignItems-1oszu61 rn-backgroundColor-44z8sh rn-borderTopColor-gv7t1l rn-borderRightColor-1itbhdy rn-borderBottomColor-1u5ghgs rn-borderLeftColor-592z09 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-5kkj8d rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1pxmb3b rn-flexBasis-7vfszb rn-flexDirection-eqz5dr rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-marginTop-1x0uki6 rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
/>
`;

exports[`Lists: List: Props Lists: List: [ prop: no props ] enzyme-shallow: snapshot: Lists: List: [ prop: no props ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderColor": "#bbb",
      "borderTopWidth": 1,
      "marginTop": 20,
    }
  }
/>
`;
